test/CMakeLists.txt

```diff
@@ -0,0 +1,3 @@
+add_test(NAME test_check_yaml_keys
+         COMMAND ${Python3_EXECUTABLE} ${CMAKE_SOURCE_DIR}/ush/check_yaml_keys.py ${CMAKE_SOURCE_DIR}/test/testinput/check_yaml_keys_ref.yaml ${CMAKE_SOURCE_DIR}/test/testinput/check_yaml_keys_test.yaml
```

Comment:
Looks good. Though, I would suggest using `${PROJECT_SOURCE_DIR}` instead.  The use of which will become clear when GDASApp is included as a project inside another project.

Same for ${PROJECT_BINARY_DIR}